src/main/java/touro/subwaymap/SubwayService.java

```diff
@@ -0,0 +1,88 @@
+package touro.subwaymap;
+
+import java.io.IOException;
+import java.util.*;
+import static java.util.Collections.*;
+import static touro.subwaymap.SubwayStations.*;
+
+public class SubwayService {
+    //clean, which belong in methods? public vs private?
+    private SubwayStations subwayStations;
+    private Station start;
+    private Station end;
+    private ArrayList<Station> unvisitedStations = new ArrayList<>();
+    private Station closestStation;
+    private final int length = 1;
+
+    //where should this kind of setup be?
+    // if not here,getSubwayRoute would need a SubwayStations object as a param? (not intuitive)
+    //create constructor for SubwayService that takes SubwayStation arg?
+    public void initialize() throws IOException {
+        JSONReader jsonReader = new JSONReader();
+        subwayStations = jsonReader.readSubwayStationJSON();
+        Map<String, ArrayList<Integer>> lineStationMap = jsonReader.readSubwayLinesJSON();
+        SubwayDataManager subwayDataManager = new SubwayDataManager();
+        subwayDataManager.processSubwayData(subwayStations, lineStationMap);
+    }
+
+    public ArrayList<Station> getSubwayRoute (double [] startLocation, double [] endLocation) {//what params would really be passed?
+        start = findNearestStation(startLocation[0],startLocation[1]);
+        end = findNearestStation(endLocation[0],endLocation[1]);
+        ArrayList<Station> shortestRoute = findShortestRoute();
+        return shortestRoute; //append final station? rewrite loop to include end/ exclude start?
+    }
+
+    private Station findNearestStation(double longitude, double latitude) {
```

Comment:
usually it's latitude, longitude

Reply:
In the json file, it gives the longitude of each station first. Should I still change my methods to match convention?